src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -352,22 +352,15 @@ void SQLStorage::storeMetadata(const Uptane::MetaPack& metadata) {
     return;
   }
 
-  if (db.exec("DELETE FROM meta;", nullptr, nullptr) != SQLITE_OK) {
+  if (db.exec("DELETE FROM " + tablename + ";", nullptr, nullptr) != SQLITE_OK) {
     LOG_ERROR << "Can't clear meta: " << db.errmsg();
     return;
   }
 
-  std::vector<std::string> jsons;
-  jsons.push_back(Json::FastWriter().write(metadata.director_root.original()));
-  jsons.push_back(Json::FastWriter().write(metadata.director_targets.original()));
-  jsons.push_back(Json::FastWriter().write(metadata.image_root.original()));
-  jsons.push_back(Json::FastWriter().write(metadata.image_targets.original()));
-  jsons.push_back(Json::FastWriter().write(metadata.image_timestamp.original()));
-  jsons.push_back(Json::FastWriter().write(metadata.image_snapshot.original()));
-
   auto statement = db.prepareStatement<SQLBlob, SQLBlob, SQLBlob, SQLBlob, SQLBlob, SQLBlob>(
-      "INSERT INTO meta VALUES (?,?,?,?,?,?);", SQLBlob(jsons[0]), SQLBlob(jsons[1]), SQLBlob(jsons[2]),
-      SQLBlob(jsons[3]), SQLBlob(jsons[4]), SQLBlob(jsons[5]));
+      "INSERT INTO " + tablename + " VALUES (?,?,?,?,?,?);", SQLBlob(metadata.director_root),
```

Comment:
Again here with the string concatenation for SQL.

Reply:
Same here.